src/App.js

```diff
@@ -1,18 +1,22 @@
 import { useEffect } from "react";
-import { useReducerContext } from './context/context';
+import { useProductContext } from './context/context';
 import { RoutesPath } from './Routes'
 import { getProductsData } from './serverCalls/getProductsData';
 function App() {
-  const { dispatch } = useReducerContext();
+  const { dispatch } = useProductContext();
   async function loadProductsData() {
+    dispatch({ type: "LOADER", payload: true });
     const InitialProductsData = await getProductsData();
     if (InitialProductsData) {
       dispatch({ type: "INITIAL PRODUCTS", payload: InitialProductsData })
+      dispatch({ type: "LOADER", payload: false });
     }
+    const encodedToken = localStorage.getItem("token");
+    dispatch({ type: "USERSIGNED", payload: encodedToken })
   }
   useEffect(() => {
     loadProductsData();
```

Comment:
suggestion: this is also fine but here you can use iife in useEffect itself instead normal function

Reply:
done